### ISSUE TITLE
Moving platform and 3D issues

### ISSUE THREAD

**Issue by fabriceci** [godotengine/godot#36483]
**Godot version:**
**OS/device including version:**
All

**Issue description:**

It seems to be an already know issue because it was mostly fixed in 3.1 for the 2D part and [this blog post](https://godotengine.org/article/godot-31-will-get-many-improvements-kinematicbody) announces that "they will be ported to the 3D engine". However as I can't find any issue related to this on Github, so I created this one.

In summary in vertical movement, the body "enter" in the platform, and in horizontal movement, there is inertia when changing direction. (strangely `move_and_slide_with_snap` worsens the issue)

**Minimal reproduction project:**
[here](https://mega.nz/#!pEsHGIwR!urA8VYoQbDlI_L3QVC8TnzAgjQTSDadLmib1d9WCgSk)

**Comment by madmiraal** [comment]
It's unclear to me what the issue is.
- The functions mentioned in the [blog](https://godotengine.org/article/godot-31-will-get-many-improvements-kinematicbody) you refer to, all do exist in 3D.
- In your example project I don't see the body entering the platform.
- I also don't understand what you mean by there being inertia when changing direction.

I would suggest closing this issue and creating a separate issue for each problem. Then, clearly explain what the current behaviour is and what the expected behaviour is. Also, provide an minimal reproduction project that clearly shows the problem. Finally, you should upload the minimal project directly to Github instead of using a link to an external storage provider.

**Comment by fabriceci** [comment]
> It's unclear to me what the issue is.
> I also don't understand what you mean by there being inertia when changing direction.
> The functions mentioned in the blog you refer to, all do exist in 3D.

Thanks to the interest @madmiraal. Indeed, I should have explained the issue instead of link a blog post. The issue is: "When a body is on a moving platform, he doesn't follow the platform's movements.”

If you read the blog post, in the part, "SYNC TO PHYSICS", you will understand what I called inertia in the gif. In 2D, 3.1 fix that with the "sync to physics" checkbox, but this checkbox doesn't exist in KinematicBody3D.

When a body is on a moving platform, you expect that the body follows the platform without moving/sliding/inertia (see the blog post).

> In your example project I don't see the body entering the platform.
> creating a separate issue for each problem.

If you observe the cube movement, you will see it's not linear but step by step because it collides. That's the same issue, because the body doesn't follow properly, it collides.

**Comment by madmiraal** [comment]
> Add sync to physics checkbox to 3D KinematicBody

That would make this a feature request and not a bug. Feature requests should be made [here](https://github.com/godotengine/godot-proposals).

**Comment by fabriceci** [comment]
> That would make this a feature request and not a bug. 

I'm not agree, it's a possible solution.

Either a KinematicBody follow the platform's movement, or not. But currently, it follows the platform but it doesn't do it well, which seems to me to be a bug. You can see why in the blog post I linked about the same issue in 2D:

> When a KinematicBody is moved around, it automatically computes its linear and angular velocity [...] This makes KinematicBody also very useful for moving platforms [...] its motion was always a frame ahead than the physics, so the character appears to slowly slide over it. To compensate with this, a new option "Sync to Physics" was added. 

If I'm wrong, feel free to close the issue.

**Comment by madmiraal** [comment]
@fabriceci I agree it's a desirable feature, but it's still a feature request; even if a blog post suggested it would be implemented in the future.

All feature and improvement proposals for the Godot Engine are now being discussed and reviewed in the dedicated [**Godot Improvement Proposals (GIP)**](https://github.com/godotengine/godot-proposals) ([godotengine/godot-proposals](https://github.com/godotengine/godot-proposals)) issue tracker. The GIP tracker has a detailed issue template designed so that proposals include all the relevant information to start a productive discussion and help the community assess the validity of the proposal for the engine.

The main ([godotengine/godot](https://github.com/godotengine/godot)) tracker is now solely dedicated to bug reports and Pull Requests, enabling contributors to have a better focus on bug fixing work. Therefore, all feature proposals on the main issue tracker are being closed.

Assuming you are interested in this feature proposal, please [**open a new proposal on the GIP tracker**](https://github.com/godotengine/godot-proposals/issues/new/choose) following the given issue template (after checking that it doesn't exist already). Be sure to reference this closed issue if it includes any relevant discussion (which you are also encouraged to summarize in the new proposal).

**Comment by fabriceci** [comment]
@madadam No problem, it's [done](https://github.com/godotengine/godot-proposals/issues/961)
